tests/e2e/test_utils.go

```diff
@@ -182,3 +183,253 @@ func GetSampleCR(t *testing.T) *v1alpha1.LlamaStackDistribution {
 
 	return distribution
 }
+
+// checkLlamaStackDistributionStatus helps identify if the custom resource reached the expected state during test execution.
+func checkLlamaStackDistributionStatus(t *testing.T, testenv *TestEnvironment, namespace, name string) {
+	t.Helper()
+
+	llsDistro := &v1alpha1.LlamaStackDistribution{}
+	err := testenv.Client.Get(testenv.Ctx, client.ObjectKey{Namespace: namespace, Name: name}, llsDistro)
+	if err != nil {
+		t.Logf("⚠️  Error getting LlamaStackDistribution: %v", err)
+		return
+	}
+
+	t.Logf("LlamaStackDistribution status:")
+	t.Logf("  Phase: %s", llsDistro.Status.Phase)
+	t.Logf("  Generation: %d", llsDistro.Generation)
+	t.Logf("  ResourceVersion: %s", llsDistro.ResourceVersion)
+	t.Logf("  Conditions: %+v", llsDistro.Status.Conditions)
+}
+
+// checkNamespaceEvents reveals what Kubernetes operations occurred and why they may have failed.
+func checkNamespaceEvents(t *testing.T, testenv *TestEnvironment, namespace string) {
+	t.Helper()
+
+	eventList := &corev1.EventList{}
+	err := testenv.Client.List(testenv.Ctx, eventList, client.InNamespace(namespace))
+	if err != nil {
+		t.Logf("⚠️  Error getting events: %v", err)
+		return
+	}
+
+	if len(eventList.Items) == 0 {
+		t.Log("📝 No events found in namespace")
+		return
+	}
+
+	maxEvents := 25
+	if len(eventList.Items) > maxEvents {
+		t.Logf("📝 Showing first %d events (of %d total):", maxEvents, len(eventList.Items))
+		eventList.Items = eventList.Items[:maxEvents]
+	} else {
+		t.Logf("📝 Found %d events in namespace %s:", len(eventList.Items), namespace)
+	}
+
+	for _, event := range eventList.Items {
+		t.Logf("  %s: %s (%s) - %s",
+			event.LastTimestamp.Format("15:04:05"),
+			event.Reason,
+			event.Type,
+			event.Message)
+	}
+}
+
+// checkOperatorHealth determines if test failures are caused by operator pod issues rather than application problems.
+func checkOperatorHealth(t *testing.T, testenv *TestEnvironment) {
+	t.Helper()
```

Comment:
Won't this be already validated as part of https://github.com/llamastack/llama-stack-k8s-operator/blob/main/tests/e2e/validation_test.go#L28 ? This ensures create_test is not running if operator validation fails